beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -172,9 +172,14 @@ func (s *Service) updateFinalized(ctx context.Context, cp *ethpb.Checkpoint) err
 			return err
 		}
 	}
-	if err := s.cfg.StateGen.MigrateToCold(ctx, fRoot); err != nil {
-		return errors.Wrap(err, "could not migrate to cold")
-	}
+	go func() {
+		// We do not pass in the parent context from the method as this method call
+		// is meant to be asynchronous and run in the background rather than being
+		// tied to the execution of a block.
+		if err := s.cfg.StateGen.MigrateToCold(s.ctx, fRoot); err != nil {
+			log.WithError(err).Error("could not migrate to cold")
```

Comment:
I see that `processFetchedData` already swallows errors with a log in the `ReceiveBlockBatch` path, but `ReceiveBlock` actually has a number of code paths that respond to errors that could be surfaced along this path. Have you reviewed all of those and made sure it's ok for this error to be silenced?